### PR TITLE
Add support for GCS(GoogleCloudStorage) with IAM

### LINKED ISSUE #20043: [Enhancement]: Support Google Cloud Storage(GCS) and support access GCS by IAM
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What would you like to be added?

- [ ] Remove the port in address When using GCS,  so that the minio sdk can recognize it's GCS
- [ ] Add logic to Acquire & renew token from GCP metaserver, add Authorization Header using the token acquired when accessing GCS. 

### Why is this needed?

So that user can use GCS easily as object storage for Milvus.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by haorenfsa**
Signed-off-by: shaoyue.chen <shaoyue.chen@zilliz.com>
resolves: #20043

### CODE REVIEW

File: internal/util/paramtable/service_param.go [hunk 1]
```diff
@@ -518,10 +520,24 @@ func (p *MinioConfig) initRootPath() {
...
+	var err error
+	p.UseIAM, err = strconv.ParseBool(useIAM)
+	if err != nil {
+		panic("parse bool useIAM:" + err.Error())
+	}
+}
+
+func (p *MinioConfig) initCloudProvider() {
+	p.CloudProvider = p.Base.LoadWithDefault("minio.cloudProvider", DefaultMinioCloudProvider)
+	supportedCloudProvider := map[string]bool{
```
Comment (jaime0815) [inline internal/util/paramtable/service_param.go:532]: supported cloudProvider should be declared as a constant, both chunk manager initialization and here can use it.

Comment (haorenfsa) [reply internal/util/paramtable/service_param.go]: updated

File: internal/util/paramtable/service_param.go [hunk 2]
```diff
@@ -518,10 +520,31 @@ func (p *MinioConfig) initRootPath() {
...
+	}
+}
+
+// CloudProvider supported
+const (
+	CloudProviderAWS = "aws"
+	CloudProviderGCP = "gcp"
+)
+
+var supportedCloudProvider = map[string]bool{
```
Comment (jaime0815) [inline internal/util/paramtable/service_param.go:536]: use map[string]struct{} will save memory

**Review by congqixia — COMMENTED** [review]
/approve

**Pull request merged**